airmozilla/popcorn/tests/test_views.py

```diff
@@ -54,3 +54,41 @@ def mocked_head(url, **options):
         eq_(content['title'], event.title)
         eq_(content['video_url'], location)
         eq_(response['Access-Control-Allow-Origin'], '*')
+
+    @mock.patch('requests.head')
+    def test_popcorn_data(self, rhead):
+        location = 'http://localhost'
+
+        def mocked_head(url, **options):
+            return Response(
+                '',
+                302,
+                headers={
+                    'location': location
+                }
+            )
+
+        rhead.side_effect = mocked_head
+
+        event = Event.objects.get(title='Test event')
+        event.template.name = 'this is a vid.ly video'
+        event.template.save()
+
+        event.template_environment = {'tag': 'abc123'}
+        event.save()
+
+        url = reverse('popcorn:popcorn_data')
+
+        response = self.client.get(url, {'slug': event.slug})
```

Comment:
Before this do...
```python
response = self.client.get(url)
# bad request because it's missing the slug
eq_(response.status_code, 400)

response = self.client.get(url, {'slug': ''})
# still not a valid slug value
eq_(response.status_code, 400)
```